src/metabase/lib/native.cljc

```diff
@@ -260,3 +260,10 @@
   [query :- ::lib.schema/query]
   (assert-native-query! (lib.util/query-stage query 0))
   (:engine (lib.metadata/database query)))
+
+(defmethod lib.query/can-save-method :mbql.stage/native
+  [query]
+  (every? (fn [[_k {:keys [required, default]}]]
```

Comment:
I'm not sure I agree. I think it's easier to understand the logic with this phrasing. It reads pretty naturally: "every template tag must either not be required, or have a default."

Reply:
Also it's easier to read the inner function args without the `[k v]` pair. Since we're not using the key, change the function to expect just the map `(fn [{:keys [required default]}] ...)` and make the second arg to `every?` be `(vals (template-tags query))`.

(Note also that we typically don't use commas when destructuring.)